lib/matplotlib/font_manager.py

```diff
@@ -740,7 +740,8 @@ def set_style(self, style):
         """
         if style is None:
             style = mpl.rcParams['font.style']
-        _api.check_in_list(['normal', 'italic', 'oblique'], style=style)
+        else:
+            _api.check_in_list(['normal', 'italic', 'oblique'], style=style)
```

Comment:
I'm not a fan of this kind of change. The code get's longer, harder to follow and more prone to errors.

Double checking the rcParam value here does not hurt. The only disadvantage are 300ns performance loss. But since the rcParam validation is an independent concept, we don't have a guarantee that they are synchronous. This could be an oversight. But I can also imagine cases where rcParams are used in multiple places and have a slightly broader acceptance range than what a specific function supports.

Reply:
The thing is that those 300 ns are multiplied with a huge number in many situations.

Reply:
What is huge? If huge * 300ns > 2% of the total plot time, we may start to consider, otherwise, the benefit is negligible.